app/services/supplementary-billing/process-billing-batch.service.js

```diff
@@ -1,39 +1,106 @@
 'use strict'
 
-const BillingPeriodsService = require('./billing-periods.service.js')
-const CreateBillingBatchPresenter = require('../../presenters/supplementary-billing/create-billing-batch.presenter.js')
-const InitiateBillingBatchService = require('./initiate-billing-batch.service.js')
+const BillingBatchError = require('../../errors/billing-batch.error.js')
+const BillingBatchModel = require('../../models/water/billing-batch.model.js')
+const ChargingModuleGenerateService = require('../charging-module/generate-bill-run.service.js')
+const FetchChargeVersionsService = require('./fetch-charge-versions.service.js')
+const HandleErroredBillingBatchService = require('./handle-errored-billing-batch.service.js')
+const LegacyRequestLib = require('../../lib/legacy-request.lib.js')
 const ProcessBillingPeriodService = require('./process-billing-period.service.js')
+const UnflagUnbilledLicencesService = require('./unflag-unbilled-licences.service.js')
 
-async function go (region, user) {
-  // NOTE: It will be required in the future that we cater for a range of billing periods, as changes can be back dated
-  // up to 5 years. For now though, our delivery scope is only for the 2022-2023 billing period so the final record is
-  // extracted from the `billingPeriods` array which will currently always be for the 2022-2023 billing period.
-  const billingPeriods = BillingPeriodsService.go()
-  const financialYearEndings = _financialYearEndings(billingPeriods)
+async function go (billingBatch, billingPeriods) {
+  const currentBillingPeriod = billingPeriods[billingPeriods.length - 1]
```

Comment:
Again, we should be able to use `.at(-1)` for a slightly cleaner way of getting the last item.
```suggestion
  const currentBillingPeriod = billingPeriods.at(-1)
```